### DISCUSSION TITLE
Module version

### DISCUSSION THREAD

**Discussion by leigh-pointer** [oqtane/oqtane.framework#2339] [Q&A] [answered]
I have a module that is packaged without the Server and Shared dll's because they are not needed.  When the module is installed the module version says 1.0.0 and not the version number in the Module.

```
    public class ModuleInfo : IModule
    {
        public ModuleDefinition ModuleDefinition => new ModuleDefinition
        {
            Name = "Studio-Elf MyModule",
            Description = "Studio-Elf MyModule]",
            Version = "3.1.4",
            ReleaseVersions = "1.0.0,1.1.0,3.1.0,3.1.1,3.1.4",
            PackageName = "StudioElf.MyModule" 
        };
    }
```

**Comment by sbwalker** [comment]
Do you razor components for this module have the exact same namespace as your IModule class?

Razor components:

@namespace StudioElf.MyModule

IModule class

namespace StudioElf.MyModule

**Comment by leigh-pointer** [comment]
Hi Shaun, yes I can confirm the namespace is correct in both.
________________________________

**Reply by sbwalker** [reply]
I would not be able to reproduce unless I had your custom module code.

**Comment by sbwalker** [comment]
@leigh-pointer I figured out what is happening here. The ModuleDefinition table has a Version column which is initialized based on the Version property in IModule. However, on upgrade the logic for updating the Version column is currently based on a module having a ServerManagerType defined (which is required in order to perform any upgrade logic such as database migrations). In your case the module is purely client-side and has no server-side requirements - so it has no ServerManagerType. As a result the Version column in the ModuleDefinition table is never updated to the latest value of the Version property in IModule. I believe it will be possible to refactor the module upgrade logic to not have a requirement on ServerManagerType for these types of scenarios.

**Reply by leigh-pointer** [reply]
@sbwalker I thought there would be away and will be happy to help

**Comment by sbwalker** [comment, marked as answer]
@leigh-pointer  #2341 should resolve this issue

**Reply by leigh-pointer** [reply]
@sbwalker that works great thanks
